File: platform/polyglot/fetch_http_cache_response/mjs/src/exceptions.ts
/**
 * Exception hierarchy for fetch-http-cache-response.
 */

export class FetchCacheError extends Error {
  cause?: Error;
  constructor(message: string, cause?: Error) {
    super(message);
    this.name = "FetchCacheError";
    this.cause = cause;
  }
}

export class FetchCacheConfigError extends FetchCacheError {
  constructor(message: string, cause?: Error) {
    super(message, cause);
    this.name = "FetchCacheConfigError";
  }
}

export class FetchCacheAuthError extends FetchCacheError {
  constructor(message: string, cause?: Error) {
    super(message, cause);
    this.name = "FetchCacheAuthError";
  }
}

export class FetchCacheNetworkError extends FetchCacheError {
  constructor(message: string, cause?: Error) {
    super(message, cause);
    this.name = "FetchCacheNetworkError";
  }
}

export class FetchCacheStorageError extends FetchCacheError {
  constructor(message: string, cause?: Error) {
    super(message, cause);
    this.name = "FetchCacheStorageError";
  }
}

export class FetchCacheTimeoutError extends FetchCacheError {
  constructor(message: string, cause?: Error) {
    super(message, cause);
    this.name = "FetchCacheTimeoutError";
  }
}

File: platform/polyglot/fetch_http_cache_response/py/fetch_http_cache_response/exceptions.py
"""Exception hierarchy for fetch_http_cache_response."""


class FetchCacheError(Exception):
    """Base exception for all fetch-cache errors."""

    def __init__(self, message: str, cause: Exception | None = None):
        super().__init__(message)
        self.__cause__ = cause


class FetchCacheConfigError(FetchCacheError):
    """Invalid configuration."""


class FetchCacheAuthError(FetchCacheError):
    """Token refresh or auth failure."""


class FetchCacheNetworkError(FetchCacheError):
    """HTTP transport error (wraps httpx/undici errors)."""


class FetchCacheStorageError(FetchCacheError):
    """S3 read/write failure."""


class FetchCacheTimeoutError(FetchCacheError):
    """Request timeout."""
